src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationStrategies/DelegationValidationStrategy.cs

```diff
@@ -344,84 +371,42 @@ public bool IsValidCallNode(CallNode node, TexlBinding binding, OperationCapabil
+            if (!isPure && !binding.Features.HasFlag(Features.AllowImpureNodeDelegation))
             {
-                var callInfo = binding.GetInfo(callNode);
-                if (callInfo != null)
-                {
-                    if (binding.IsBlockScopedConstant(node) || (callInfo.Function is AsTypeFunction))
-                    {
-                        // AsType is delegable despite being async
-                        return true;
-                    }
-                }
-                else
-                {
-                    return false;
-                }
+                TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.ImpureNode, node, binding, Function, DelegationTelemetryInfo.CreateImpureNodeTelemetryInfo(node, binding));
+                return false;
             }
 
-            // Async predicates and impure nodes are not supported.
-            // Let CallNodes for User() be marked as being Valid to allow
-            // expressions with User() calls to be delegated
-            if (!IsUserCallNodeDelegable(node, binding) && (isAsync || !isPure))
+            // If the feature is enabled, enable delegation for
+            // async call, first name and dotted name nodes.
+            if (binding.Features.HasFlag(Features.AllowAsyncDelegation)
+                && ((node is CallNode) || (node is FirstNameNode) || (node is DottedNameNode)))
             {
-                var telemetryMessage = string.Format(CultureInfo.InvariantCulture, "Kind:{0}, isAsync:{1}, isPure:{2}", node.Kind, isAsync, isPure);
-                SuggestDelegationHintAndAddTelemetryMessage(node, binding, telemetryMessage);
-
-                if (isAsync)
-                {
-                    TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.AsyncPredicate, node, binding, Function);
-                }
-
-                if (!isPure)
-                {
-                    TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.ImpureNode, node, binding, Function, DelegationTelemetryInfo.CreateImpureNodeTelemetryInfo(node, binding));
-                }
-
-                return false;
+                return true;
             }
 
-            return true;
-        }
-
-        private bool IsUserCallNodeDelegable(TexlNode node, TexlBinding binding)
-        {
-            if ((node is DottedNameNode)
-                && (node.AsDottedName().Left is CallNode)
-                && (binding.GetInfo(node.AsDottedName().Left.AsCall()).Function is ICustomDelegationFunction customDelFunc)
-                && customDelFunc.IsUserCallNodeDelegable())
+            if (isAsync && !binding.Features.HasFlag(Features.AllowAsyncDelegation))
             {
-                return true;
+                TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.AsyncPredicate, node, binding, Function);
             }
```

Comment:
I would simplify this even

if (binding.Features.HasFlag(Features.AllowAsyncDelegation))
{
    if(((node is CallNode) || 
         (node is FirstNameNode) || 
         (node is DottedNameNode)))
      { 
           return true;
       }
}
else
{
      if(isAsync)               TrackingProvider.Instance.SetDelegationTrackerStatus(DelegationStatus.AsyncPredicate, node, binding, Function);
}

Reply:
Why it would be the sane logic unless I am missing something.